Performance/QPSBenchmark/Sources/QPSBenchmark/Runtime/Async/AsyncQPSClientImpl.swift

```diff
@@ -0,0 +1,293 @@
+        throw GRPCStatus(code: .invalidArgument, message: "Byte buffer not supported.")
+      case let .simpleParams(simpleParams):
+        var result = Grpc_Testing_SimpleRequest()
+        result.responseType = .compressable
+        result.responseSize = simpleParams.respSize
+        result.payload.type = .compressable
+        let size = Int(simpleParams.reqSize)
+        let body = Data(count: size)
+        result.payload.body = body
+        return result
+      case .complexParams:
+        throw GRPCStatus(
+          code: .invalidArgument,
+          message: "Complex params not supported."
+        )
+      }
+    } else {
+      // Default - simple proto without payloads.
+      var result = Grpc_Testing_SimpleRequest()
+      result.responseType = .compressable
+      result.responseSize = 0
+      result.payload.type = .compressable
+      return result
+    }
+  }
+
+  /// Class to manage a channel.  Repeatedly makes requests on that channel and records what happens.
+  private final class ChannelRepeater: @unchecked Sendable {
+    private let channel: GRPCChannel
+    private let eventLoop: EventLoop
+    private let maxPermittedOutstandingRequests: Int
+
+    private let stats: StatsWithLock
+
+    /// Succeeds after a stop has been requested and all outstanding requests have completed.
+    private let stopComplete: EventLoopPromise<Void>
+
+    private let running = ManagedAtomic<Bool>(false)
+
+    private let requestMaker: RequestMakerType
+
+    init(
+      target: ConnectionTarget,
+      requestMessage: Grpc_Testing_SimpleRequest,
+      config: Grpc_Testing_ClientConfig,
+      eventLoop: EventLoop
+    ) {
+      self.eventLoop = eventLoop
+      // 'try!' is fine; it'll only throw if we can't make an SSL context
+      // TODO: Support TLS if requested.
+      self.channel = try! GRPCChannelPool.with(
+        target: target,
+        transportSecurity: .plaintext,
+        eventLoopGroup: eventLoop
+      )
+
+      let logger = Logger(label: "ChannelRepeater")
+      let client = Grpc_Testing_BenchmarkServiceAsyncClient(channel: self.channel)
+      self.maxPermittedOutstandingRequests = Int(config.outstandingRpcsPerChannel)
+      self.stopComplete = eventLoop.makePromise()
+      self.stats = StatsWithLock()
+
+      self.requestMaker = RequestMakerType(
+        config: config,
+        client: client,
+        requestMessage: requestMessage,
+        logger: logger,
+        stats: self.stats
+      )
+    }
+
+    /// Launch as many requests as allowed on the channel. Must only be called once.
+    private func launchRequests() async throws {
+      // The plan here is:
+      // - store the max number of outstanding requests in an atomic
+      // - start that many requests asynchronously
+      // - when a request finishes it will either start a new request or decrement the
+      //   the atomic counter (if we've been told to stop)
+      // - if the counter drops to zero we're finished.
```

Comment:
I think we can just remove this comment: it's no longer accurate and the code is more straightforward than it was previously so not sure it needs it.